### PR TITLE
[JBPM-9407] Spring Boot - Multiple Data Sources and Pluggable Variable Persistence

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9407

Overwrite for spring boot for adding custom entity managers

depends on:
https://github.com/kiegroup/jbpm/pull/1769

### CODE REVIEW

File: kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/services/SpringKModuleDeploymentService.java [hunk 1]
```diff
@@ -50,7 +53,16 @@ protected RegisterableItemsFactory getRegisterableItemsFactory(AuditEventBuilder
         return factory;
     }
 
-    
+    protected Map<String, Object> buildContextParameters(KieContainer kieContainer) {
```
Comment (gmunozfe) [inline kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/services/SpringKModuleDeploymentService.java:56]: Though it's not mandatory, perhaps it's more clear if "Override" annotation is explicitly shown.

**Review by gmunozfe — COMMENTED** [review]
Looks good, just a minor comment for enhancing clarity

File: kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/services/SpringKModuleDeploymentService.java [hunk 2]
```diff
@@ -50,7 +53,16 @@ protected RegisterableItemsFactory getRegisterableItemsFactory(AuditEventBuilder
...
+        Map<String, Object> contextParams = new HashMap<>();
+        Map<String, EntityManagerFactory> emfsFound = context.getBeansOfType(EntityManagerFactory.class);
+        for(Map.Entry<String, EntityManagerFactory> entry : emfsFound.entrySet()) {
+            contextParams.put(entry.getKey(), entry.getValue());
+        }
+        contextParams.putAll(super.buildContextParameters(kieContainer));
```
Comment (gmunozfe) [inline kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/services/SpringKModuleDeploymentService.java:62]: In the parent invocation, ("entityManagerFactory", emf) is also added, shouldn't it be redundant?

**Review by gmunozfe — COMMENTED** [review]
I've got also a question regarding parent invocation which also adds the emf.

**Review by afalhambra — COMMENTED** [review]
Looks good to me though it would be very useful to have this particular scenario in a test case.
Good work, thanks

File: kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/services/SpringKModuleDeploymentService.java (hunk 1 above)
Comment (elguardian) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/services/SpringKModuleDeploymentService.java:56]: done

File: kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/services/SpringKModuleDeploymentService.java (hunk 2 above)
Comment (elguardian) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/services/SpringKModuleDeploymentService.java:62]: not really. 
Not all deployment service use this spring for instance kie server. so the parents needs to add the entity manager.
Also we need the parameters defined in there.
There is no point to do any check away just to remove the jbpm entity manager as it cannot be override.

[the remark this answers is not in the document]
Comment (elguardian) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/services/SpringKModuleDeploymentService.java:62]: not really. 
Not all deployment service use this spring for instance kie server. so the parents needs to add the entity manager.
Also we need the parameters defined in there.
There is no point to do any check away just to remove the jbpm entity manager as it cannot be override.

**Review by afalhambra — COMMENTED** [review]
Nicely done. Will you add a test for this scenario? That's the only thing I'm missing here. Thanks

**Pull request merged**